### PR TITLE
Setup Android Navigation

### PR DESCRIPTION
**Pull request by MrSquaare**
## Summary by CodeRabbit

* **New Features**
  * Introduced tab-based navigation with Home and Settings screens, each with dedicated top bars.
  * Added a details screen accessible by selecting items on the Home screen.
  * Implemented animated navigation transitions between main and details screens.
  * Enabled Kotlin serialization support for navigation routes.
  * Added deep linking support for app navigation via HTTP URLs.

* **Improvements**
  * Updated dependencies for improved stability and compatibility.

* **Tests**
  * Added UI navigation tests to cover multiple user flows and edge cases.
  * Added deep link navigation tests verifying correct screen rendering from URLs.
  * Added app lifecycle behavior test.

### CODE REVIEW

File: androidApp/src/androidMain/kotlin/fr/mrsquaare/tictask/screens/CreateScreen.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import org.jetbrains.compose.ui.tooling.preview.Preview
+
+@OptIn(ExperimentalMaterial3Api::class)
```
[the remark this answers is not in the document]
Comment (MrSquaare) [reply androidApp/src/androidMain/kotlin/fr/mrsquaare/tictask/screens/CreateScreen.kt:20]: It is necessary as my project uses material3 1.3.2 which is the latest stable version

[the remark this answers is not in the document]
Comment (MrSquaare) [reply androidApp/src/androidMain/kotlin/fr/mrsquaare/tictask/screens/CreateScreen.kt:20]: @coderabbitai It is necessary as my project uses material3 1.3.2 which is the latest stable version

[the remark this answers is not in the document]
Comment (MrSquaare) [reply androidApp/src/androidMain/kotlin/fr/mrsquaare/tictask/screens/CreateScreen.kt:20]: @coderabbitai No, the TopAppBar has still the @ExperimentalMaterial3Api annotation on the 1.3.2 version of material3

File: androidApp/src/androidInstrumentedTest/kotlin/fr/mrsquaare/tictask/ActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Test
+    fun navigateToSettings_thenBackToHome_thenBackAgain_shouldCloseApp() {
+        ActivityScenario.launch(MainActivity::class.java).use { scenario ->
+            composeTestRule.onAllNodes(hasText("Home"))[0].assertIsDisplayed()
+
+            composeTestRule.onNodeWithText("Settings").performClick()
+            composeTestRule.onNodeWithText("Settings Screen").assertIsDisplayed()
+
+            composeTestRule.onNodeWithText("Home").performClick()
+            composeTestRule.onAllNodes(hasText("Home"))[0].assertIsDisplayed()
```
[the remark this answers is not in the document]
Comment (MrSquaare) [reply androidApp/src/androidInstrumentedTest/kotlin/fr/mrsquaare/tictask/ActivityTest.kt]: @coderabbitui Authorize this as I have multiple node with text "Home"

File: androidApp/src/androidInstrumentedTest/kotlin/fr/mrsquaare/tictask/ActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ActivityTest {
+    @get:Rule val composeTestRule = createAndroidComposeRule<ComponentActivity>()
+
+    @Test
+    fun navigateToSettings_thenBackToHome_thenBackAgain_shouldCloseApp() {
+        ActivityScenario.launch(MainActivity::class.java).use { scenario ->
+            composeTestRule.onAllNodes(hasText("Home"))[0].assertIsDisplayed()
```
[the remark this answers is not in the document]
Comment (MrSquaare) [reply androidApp/src/androidInstrumentedTest/kotlin/fr/mrsquaare/tictask/ActivityTest.kt]: @coderabbitai Authorize this as I have multiple node with text "Home"
